src/util/http/websocket/WebSocketSession.cpp

```diff
@@ -115,21 +114,21 @@ net::awaitable<void> WebSocketSession::acceptAndWait(
 net::awaitable<void> WebSocketSession::handleSession(
     QueryHub& queryHub, const QueryRegistry& queryRegistry,
     const http::request<http::string_body>& request, tcp::socket socket) {
-  // Make sure access to new websocket is on a strand and therefore thread safe
-  auto executor = co_await net::this_coro::executor;
-  auto strandExecutor =
-      executor.target<net::strand<net::io_context::executor_type>>();
-  AD_CONTRACT_CHECK(strandExecutor);
-  AD_CONTRACT_CHECK(strandExecutor->running_in_this_thread());
-  AD_CONTRACT_CHECK(socket.get_executor() == *strandExecutor);
-
   auto queryIdString = extractQueryId(request.target());
   AD_CORRECTNESS_CHECK(!queryIdString.empty());
   auto queryId = QueryId::idFromString(queryIdString);
   UpdateFetcher fetcher{queryHub, queryId};
+  auto strand = fetcher.strand();
   WebSocketSession webSocketSession{std::move(fetcher), std::move(socket),
                                     queryRegistry, std::move(queryId)};
-  co_await webSocketSession.acceptAndWait(request);
+  // There is currently no safe way of which we know that allows us to respawn
+  // the coroutine onto another thread AND make it safely cancellable at the
+  // same time. Therfore we just assert that the cancellation slot is not
+  // connected.
+  auto slot = (co_await net::this_coro::cancellation_state).slot();
+  AD_CORRECTNESS_CHECK(!slot.is_connected());
+  co_await (net::co_spawn(strand, webSocketSession.acceptAndWait(request),
+                          net::deferred));
```

Comment:
What does `net::deferred` here do exactly?

Reply:
When you directly `co_await`, then you can often use it as a more lightweight and  cheaper replacement for `use_awaitable`. But the semantics are the same.